CMakeLists.txt

```diff
@@ -291,6 +291,10 @@ if(WarpX_GPUCLOCK)
     target_compile_definitions(ablastr PUBLIC WARPX_USE_GPUCLOCK)
 endif()
 
+if(WarpX_EXTERNAL_FIELD)
```

Comment:
Just a note for now: We won't use a conditional here for something that is not dependency control.
If we need compile-time specializations, we will use a template and specialize/call the kernel that it controls from a runtime if/switch.